cmd/cli/compliance_check.go

```diff
@@ -0,0 +1,109 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	//"encoding/hex"
+	//"encoding/base64"
+	"encoding/json"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/spf13/cobra"
+	"google.golang.org/grpc"
+
+	"github.com/xuperchain/xuperunion/global"
+	"github.com/xuperchain/xuperunion/pb"
+	"github.com/xuperchain/xuperunion/utxo/txhash"
+)
+
+type GetSignCommand struct {
+	cli          *Cli
+	cmd          *cobra.Command
+	xcheckclient pb.XcheckClient
+
+	tx     string
+	host   string
+	output string
+}
+
+func NewGetSignCommand(cli *Cli) *cobra.Command {
+	c := new(GetSignCommand)
+	c.cli = cli
+	c.cmd = &cobra.Command{
+		Use:   "get",
+		Short: "get a sign from remote node.",
+		Long:  `./xchain-cli multisig get --tx ./tx.out`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx := context.TODO()
+			return c.get(ctx)
+		},
+	}
+	c.addFlags()
+	return c.cmd
+}
+
+func (c *GetSignCommand) initXcheckClient() error {
+	conn, err := grpc.Dial(c.host, grpc.WithInsecure(), grpc.WithMaxMsgSize(64<<20-1))
+	if err != nil {
+		return err
+	}
+	c.xcheckclient = pb.NewXcheckClient(conn)
+	return nil
+}
+
+func (c *GetSignCommand) XcheckClient() pb.XcheckClient {
+	c.initXcheckClient()
+	return c.xcheckclient
+}
+
+func (c *GetSignCommand) addFlags() {
+	c.cmd.Flags().StringVar(&c.tx, "tx", "./tx.out", "Serialized transaction data file")
+	c.cmd.Flags().StringVar(&c.host, "host", "localhost:6718", "host to get signature from compliance check service")
+	c.cmd.Flags().StringVar(&c.output, "output", "./compliance_check_sign.out", "Generate signature file for a transaction.")
+}
+
+func (c *GetSignCommand) get(ctx context.Context) error {
+	data, err := ioutil.ReadFile(c.tx)
+	if err != nil {
+		return errors.New("Fail to open serialized transaction data file")
+	}
+	tx := &pb.Transaction{}
+	err = proto.Unmarshal(data, tx)
+	if err != nil {
+		return errors.New("Fail to Unmarshal proto")
+	}
+
+	tx.Txid, err = txhash.MakeTransactionID(tx)
```

Comment:
Why here need a txid? at this time there is no signatures in txdata, so using MakeTransactionID cannot get the final txid.

Reply:
The client should pass TxStatus to the server.